plugins/awards-backend/src/awards.ts

```diff
@@ -2,33 +2,99 @@
  * Copyright SeatGeek
  * Licensed under the terms of the Apache-2.0 license. See LICENSE file in project root for terms.
  */
+import { TokenManager } from '@backstage/backend-common';
+import { CatalogClient } from '@backstage/catalog-client';
+import { isUserEntity } from '@backstage/catalog-model';
 import { NotFoundError } from '@backstage/errors';
 import { Award, AwardInput } from '@seatgeek/backstage-plugin-awards-common';
 import { Logger } from 'winston';
 import { AwardsStore } from './database/awards';
+import { NotificationsGateway } from './notifications/notifications';
+
+function nonNullable<T>(value: T): value is NonNullable<T> {
+  return value !== null && value !== undefined;
+}
 
 export class Awards {
   private readonly db: AwardsStore;
   private readonly logger: Logger;
+  private readonly notifications: NotificationsGateway;
+  private readonly catalogClient: CatalogClient;
+  private readonly tokenManager: TokenManager;
 
-  constructor(db: AwardsStore, logger: Logger) {
+  constructor(
+    db: AwardsStore,
+    notifications: NotificationsGateway,
+    catalogClient: CatalogClient,
+    tokenManager: TokenManager,
```

Comment:
These three new dependencies are only used by the `notifyNewRecipients` function, but now the entire class is untestable unless these are all injected/mocked.  Should we decouple this with one additional abstraction layer (perhaps called `AwardsNotifier` or something) between the awards and the gateway so we can inject a single `AwardNotifier` and keep that logic isolated from awards management? WDYT?

Reply:
yep i think that makes sense. `AwardsNotifier` could also have some smarter logic that i kept out from bloating the `Awards` class like fanning out to multiple notification gateways and skipping catalog fetching if there are no notification gateways (rather than the nullnotificationgateway)

Reply:
done in fb4f4f0.. how's that look?

Reply:
lol yeah i love that. i think it's the [composite pattern](https://en.wikipedia.org/wiki/Composite_pattern)?